From b82d395a110f74a3b6c1423a5a0eb6ead09cc593 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 28 Apr 2021 22:54:48 +0300
Subject: [PATCH] [SW-40869] lkd: fix potential bug in reset_state

The reset_state command implementation is not resetting the commands
state machine.
Fix this potential bug - reset the commands state machine.

Change-Id: Ic3dd0580f90258ee5d46da384ec327c20965cb98
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/preboot/module_lkd.c b/preboot/module_lkd.c
index 88e9d58..a3a1418 100644
--- a/preboot/module_lkd.c
+++ b/preboot/module_lkd.c
@@ -132,6 +132,14 @@
 	return 1;
 }
 
+static void module_lkd_reset_state(void)
+{
+	sts.exec_res = NOT_FOUND;
+	sts.cmd.val = 0;
+	sts.wfc_cmd.val = 0;
+	sts.do_cmd.val = 0;
+}
+
 static void module_lkd_run_fast_noop(void)
 {
 	int err;
@@ -146,7 +154,7 @@
 		module_lkd_send_status(COMMS_STS_NOOP, COMMS_SRAM, 0);
 		break;
 	case COMMS_RST_STATE:
-		sts.exec_res = NOT_FOUND;
+		module_lkd_reset_state();
 		module_lkd_send_status(COMMS_STS_OK, COMMS_SRAM, 0);
 		break;
 	case COMMS_EXEC:
